react-frontend/src/pages/MakeOfferPage.js

```diff
@@ -1,37 +1,101 @@
   const { listingID } = useParams();
 
+  useEffect(() => {
+    const fetchUserProfile = async () => {
+      const token = localStorage.getItem('token');
+      if (!token) {
+        navigate('/login');
+        return;
+      }
+
+      try {
+        const response = await axios.get(`${process.env.REACT_APP_BACKEND_LINK}/users/profile`, {
+          headers: { 'Authorization': `Bearer ${token}` },
+        });
+        setBuyer(response.data);
+      } catch (error) {
+        console.error('Failed to fetch user data:', error);
+        navigate('/login');
+      }
+    };
+
+    fetchUserProfile();
+  }, [navigate]);
+
   const handleMakeOffer = async () => {
+    if (!buyer) {
+      console.error("Buyer information is not available.");
+      return;
+    }
+
     try {
-        // API call to create the chat and send the initial offer message
-        const response = await axios.post(`/api/offers/${listingID}`, { offer });
-        const chatID = response.data.chatID;
-        navigate(`/chat/${chatID}`);
+      const { data: listing } = await axios.get(`/api/listings/${listingID}`);
+      const { data: seller } = await axios.get(`/api/users/${listing.userId}`);
+  
+      Talk.ready.then(() => {
+        const me = new Talk.User({
+          id: buyer.userId.toString(),
+          name: buyer.name,
+          email: buyer.email,
+          photoUrl: buyer.photoUrl,
+          welcomeMessage: "Hi, I would like to make an offer!"
+        });
+  
+        const other = new Talk.User({
+          id: seller.userId.toString(),
+          name: seller.name,
+          email: seller.email,
+          photoUrl: seller.photoUrl,
+          welcomeMessage: "Hello, are you interested in my listing?"
+        });
+  
+        const session = new Talk.Session({
+          appId: TALKJS_APP_ID,
+          me: me
+        });
+  
+        const conversationId = Talk.oneOnOneId(me, other);
+        const conversation = session.getOrCreateConversation(conversationId);
+        conversation.setParticipant(me);
+        conversation.setParticipant(other);
+        conversation.setAttributes({
+          subject: `Offer for ${listing.title}`
+        });
+  
+        const chatbox = session.createChatbox(conversation);
+        chatbox.mount(document.getElementById("talkjs-container"));
+  
+        navigate(`/chat/${conversationId}`);
+      });
     } catch (error) {
-        console.error('Error making offer:', error);
+      console.error('Error making offer:', error);
     }
   };
 
   const handleInputChange = (event) => {
     const { value } = event.target;
-    const numbers = value.replace(/[^0-9]/g, '');  // Strip non-numeric characters
-    setOffer(numbers);  // Set the numbers directly as the offer
+    const numbers = value.replace(/[^0-9]/g, '');
+    setOffer(numbers);
   };
 
   const formatNumber = (number) => {
-    // Convert number to a string and format as USD currency without cents
     return new Intl.NumberFormat('en-US', {
```

Comment:
Visual Suggestion: Add a loading state using LoadingSpinner for rendering HTML

Reply:
that is a great suggestion. I will look into it